gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -160,6 +178,11 @@
 
   private static final ListFileOptions LIST_OPTIONS =
       ListFileOptions.DEFAULT.toBuilder().setFields(OBJECT_FIELDS).build();
+  /** Bhagyaa-s**/
+  private static final int DEFAULT_MAX_THREADS = 64; //16
```

Comment:
Can we reduce the default to 16, and add an option to override this via property